src/platform/orchestrator/src/MpiApi.cpp

```diff
@@ -283,19 +282,25 @@ static void MpiGetRequest(const Request& request, Response& response)
                 std::string payloadString = std::string(payload, payloadSizeBytes);
 
                 response.m_status = ((status == MPI_OK) ? StatusCode::OK : StatusCode::BAD_REQUEST);
-                response.m_headers[contentLength] = std::to_string(payloadString.size());
                 response.m_body = payloadString;
             }
             else
             {
-                OsConfigLogError(PlatformLog::Get(), "Invalid MPI get request");
+                OsConfigLogError(PlatformLog::Get(), "Invalid MpiGet request");
+                response.m_status = StatusCode::BAD_REQUEST;
             }
         }
         else
         {
-            OsConfigLogError(PlatformLog::Get(), "Invalid MPI get request");
+            OsConfigLogError(PlatformLog::Get(), "Invalid MpiGet request body: %s", request.m_body.c_str());
```

Comment:
>OsConfigLogError(PlatformLog::Get(), "Invalid MpiGet request body: %s", request.m_body.c_str()); [](http://example.com/codeflow?start=12&length=96)

May not be good to log this in full unless full logging.

But also log for normal logging, just w/o payload values.